src/overlays/actors/ovl_En_Dt/z_en_dt.c

```diff
@@ -28,59 +95,714 @@ ActorInit En_Dt_InitVars = {
     /**/ EnDt_Draw,
 };
 
-// static ColliderCylinderInit sCylinderInit = {
-static ColliderCylinderInit D_80BEB29C = {
-    { COLTYPE_NONE, AT_NONE, AC_NONE, OC1_ON | OC1_TYPE_ALL, OC2_TYPE_2, COLSHAPE_CYLINDER, },
-    { ELEMTYPE_UNK0, { 0x00000000, 0x00, 0x00 }, { 0xF7CFFFFF, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_NONE, OCELEM_ON, },
+static u16 sTextIds[] = { 0x2ABD, 0x2ABE, 0x2ABF, 0x2AC0, 0x2ABE, 0x2AC1, 0x2AC2, 0x2AC3, 0x2AC4, 0x2ABB,
+                          0x2ABC, 0x2AC6, 0x2AC7, 0x2AC8, 0x2AC9, 0x2ACA, 0x2ACB, 0x2ACC, 0x2ACD, 0x2ACE,
+                          0x2ACF, 0x2AD0, 0x2AD1, 0x2AC5, 0x2AD3, 0x2AD4, 0x2AD2, 0x0000 };
+
+// The code uses EnDt::textIdIndex as an index into this lookup table
+static s32 sCutsceneFocusTargetTable[] = {
+    /* 00 */ EN_DT_CS_FOCUS_TARGET_BAISEN, /* [2ABD] "Most of the townsfolk already sheltered" */
+    /* 01 */ EN_DT_CS_FOCUS_TARGET_MAYOR,  /* [2ABE] "Ahh...hmm...well..." */
+    /* 02 */ EN_DT_CS_FOCUS_TARGET_MUTO,   /* [2ABF] "You cowards! The moon won't fall" */
+    /* 03 */ EN_DT_CS_FOCUS_TARGET_MUTO,   /* [2AC0] "Isn't that right Mayor?" */
+    /* 04 */ EN_DT_CS_FOCUS_TARGET_MAYOR,  /* [2ABE] "Ahh...hmm...well..." */
+    /* 05 */ EN_DT_CS_FOCUS_TARGET_BAISEN, /* [2AC1] "You serious Mutoh? Missed the huge rock?" */
+    /* 06 */ EN_DT_CS_FOCUS_TARGET_MAYOR,  /* [2AC2] "Mmm...Hmm..." */
+    /* 07 */ EN_DT_CS_FOCUS_TARGET_MUTO,   /* [2AC3] "If the soldiers wish to run, then abandon us"   */
+    /* 08 */ EN_DT_CS_FOCUS_TARGET_MAYOR,  /* [2AC4] "Let's not bring my wife into this" */
+
+    /* 09 */ EN_DT_CS_FOCUS_TARGET_BAISEN, /* [2ABB] "All must take refuge!!!" */
+    /* 10 */ EN_DT_CS_FOCUS_TARGET_MUTO,   /* [2ABC] "On with the carnival!!!" */
+
+    /* 11 */ EN_DT_CS_FOCUS_TARGET_BAISEN, /* [2AC6] "Ah!" */
+    /* 12 */ EN_DT_CS_FOCUS_TARGET_MAYOR,  /* [2AC7] "Ah!" */
+    /* 13 */ EN_DT_CS_FOCUS_TARGET_MUTO,   /* [2AC8] "Ah!" */
+    /* 14 */ EN_DT_CS_FOCUS_TARGET_MAYOR,  /* [2AC9] "The Couple's Mask. Someone got married" */
+    /* 15 */ EN_DT_CS_FOCUS_TARGET_MUTO,   /* [2ACA] "I wonder if my wife fled" */
+    /* 16 */ EN_DT_CS_FOCUS_TARGET_BAISEN, /* [2ACB] "What did you just say!?" */
+    /* 17 */ EN_DT_CS_FOCUS_TARGET_MAYOR,  /* [2ACB] "We're all worried, let's end the meeting" */
+    /* 18 */ EN_DT_CS_FOCUS_TARGET_BAISEN, /* [2ACD] "But Mayor!" */
+    /* 19 */ EN_DT_CS_FOCUS_TARGET_MAYOR,  /* [2ACE] "Stay or run, that's up to people to decide" */
+    /* 20 */ EN_DT_CS_FOCUS_TARGET_MUTO,   /* [2ACF] "..!" */
+    /* 21 */ EN_DT_CS_FOCUS_TARGET_MAYOR,  /* [2AD0] "Thanks for helping me end that. Here's your reward" */
+    /* 22 */ EN_DT_CS_FOCUS_TARGET_MAYOR,  /* [2AD1] "I'm an unreliable mayor, but I want to protect my family" */
+
+    /* 23 */ EN_DT_CS_FOCUS_TARGET_MAYOR /* [2AC5] "You take refuge, too." */
+};
+
+typedef struct {
+    s16 D_80BEB268[1];
+    s16 D_80BEB26A[25];
+} CutsceneIndexTable;
+
+static CutsceneIndexTable CsIndexTable = { { 0x2ABB },
```

Comment:
This table is a bit strange to me. It seems like the developers offset the initial data pointer depending on whether they wanted the cutscene ID index or the string ID, and then incremented by 2. I couldn't find a more appropriate data representation that still resulting in a match.

Open to suggestions here.

Reply:
Yep, makes perfect sense! I seriously over-complicated this because I didn't realize that indexing into a single dimension array with an offset of one like that would generate two distinct addresses in the final asm, so I thought they had to be separate symbols. Thanks for helping me understand that.